src/parser/index.ts

```diff
@@ -374,11 +379,10 @@ const otherExpr = seq(
 )((first, rest) =>
   rest.reduce((acc, val) => OtherExprRhs.createExpression(acc, val), first)
 )
-const existsOrOtherExpr = oneOf(existsExpr, otherExpr)
 
 const comparisonExpr = makeBinaryOp(
   oneOfOperators('<', '>', '=', '<=', '>=', '<>'),
-  existsOrOtherExpr
+  oneOf(existsExpr, anyExpr, otherExpr)
```

Comment:
It seems to me that this would allow the following on the syntax level:

```
select any(...) = any(...)
```

which is not correct, although

```
select 1 = any (...) = any (...)
```

seems to be correct.

In other words, expressions of the form `lhs op any/some/all (...)` should be have their own "level" in the operator precedence hierarchy.